### PR TITLE
executor: let flush privileges do nothing when `skip-grant-table` is configured

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

When skip-grant-table is enabled, privilege handle is not initialized, calling flush privileges
would meet nil pointer panic

```
mysql> flush privileges;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```

```
github.com/pingcap/tidb/server.(*clientConn).Run.func1
	/root/go/src/github.com/pingcap/tidb/server/conn.go:593
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:522
runtime.panicmem
	/usr/local/go/src/runtime/panic.go:82
runtime.sigpanic
	/usr/local/go/src/runtime/signal_unix.go:390
github.com/pingcap/tidb/privilege/privileges.(*Handle).Update
	/root/go/src/github.com/pingcap/tidb/privilege/privileges/cache.go:979
github.com/pingcap/tidb/executor.(*SimpleExec).executeFlush
	/root/go/src/github.com/pingcap/tidb/executor/simple.go:822
github.com/pingcap/tidb/executor.(*SimpleExec).Next
	/root/go/src/github.com/pingcap/tidb/executor/simple.go:74
github.com/pingcap/tidb/executor.Next
	/root/go/src/github.com/pingcap/tidb/executor/executor.go:190
github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor
	/root/go/src/github.com/pingcap/tidb/executor/adapter.go:388
github.com/pingcap/tidb/executor.(*ExecStmt).Exec
	/root/go/src/github.com/pingcap/tidb/executor/adapter.go:265
github.com/pingcap/tidb/session.runStmt
	/root/go/src/github.com/pingcap/tidb/session/tidb.go:219
github.com/pingcap/tidb/session.(*session).executeStatement
	/root/go/src/github.com/pingcap/tidb/session/session.go:939
github.com/pingcap/tidb/session.(*session).execute
	/root/go/src/github.com/pingcap/tidb/session/session.go:1037
github.com/pingcap/tidb/session.(*session).Execute
	/root/go/src/github.com/pingcap/tidb/session/session.go:966
github.com/pingcap/tidb/session.execRestrictedSQL
	/root/go/src/github.com/pingcap/tidb/session/session.go:745
github.com/pingcap/tidb/session.(*session).ExecRestrictedSQL
	/root/go/src/github.com/pingcap/tidb/session/session.go:700
github.com/pingcap/tidb/domain.(*Domain).NotifyUpdatePrivilege
	/root/go/src/github.com/pingcap/tidb/domain/domain.go:1047
github.com/pingcap/tidb/executor.(*SimpleExec).executeSetPwd
	/root/go/src/github.com/pingcap/tidb/executor/simple.go:789
github.com/pingcap/tidb/executor.(*SimpleExec).Next
	/root/go/src/github.com/pingcap/tidb/executor/simple.go:88
github.com/pingcap/tidb/executor.Next
	/root/go/src/github.com/pingcap/tidb/executor/executor.go:190
github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor
	/root/go/src/github.com/pingcap/tidb/executor/adapter.go:388
github.com/pingcap/tidb/executor.(*ExecStmt).Exec
	/root/go/src/github.com/pingcap/tidb/executor/adapter.go:265
github.com/pingcap/tidb/session.runStmt
	/root/go/src/github.com/pingcap/tidb/session/tidb.go:219
github.com/pingcap/tidb/session.(*session).executeStatement
	/root/go/src/github.com/pingcap/tidb/session/session.go:939
github.com/pingcap/tidb/session.(*session).execute
	/root/go/src/github.com/pingcap/tidb/session/session.go:1037
github.com/pingcap/tidb/session.(*session).Execute
	/root/go/src/github.com/pingcap/tidb/session/session.go:966
github.com/pingcap/tidb/server.(*TiDBContext).Execute
	/root/go/src/github.com/pingcap/tidb/server/driver_tidb.go:246
github.com/pingcap/tidb/server.(*clientConn).handleQuery
	/root/go/src/github.com/pingcap/tidb/server/conn.go:1158
github.com/pingcap/tidb/server.(*clientConn).dispatch
	/root/go/src/github.com/pingcap/tidb/server/conn.go:885
github.com/pingcap/tidb/server.(*clientConn).Run
	/root/go/src/github.com/pingcap/tidb/server/conn.go:644
github.com/pingcap/tidb/server.(*Server).onConn
	/root/go/src/github.com/pingcap/tidb/server/server.go:438
```

### What is changed and how it works?

Make `flush privileges` return directly if `skip-grant-table` is enabled.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @zz-jason @jackysp

File: executor/simple_test.go [hunk 1]
```diff
@@ -394,6 +394,10 @@ func (s *testSuite3) TestFlushPrivileges(c *C) {
 	// After flush.
 	_, err = se.Execute(ctx, `SELECT Password FROM mysql.User WHERE User="testflush" and Host="localhost"`)
 	c.Check(err, IsNil)
+
+	config.GetGlobalConfig().Security.SkipGrantTable = true
+	tk.MustExec("FLUSH PRIVILEGES")
```
Comment (zz-jason) [inline executor/simple_test.go]: I run this test in the master branch, it passed. I think it should result in panic?

Comment (tiancaiamao) [reply executor/simple_test.go]: It should panic!
Try to config `skip-grant-table = true` and start a TiDB (master branch), calling "flush privileges" through the mysql client you would reproduce the problem.

Comment (zz-jason) [reply executor/simple_test.go]: No, I mean I run this unit test, it passed. Not in a mysql client.

**Pull request merged**